runtime_scan/pkg/provider/docker/client.go

```diff
@@ -0,0 +1,394 @@
+	if containerID != "" {
+		return containerID, nil
+	}
+
+	pl, err := c.dockerClient.ImagePull(ctx, config.ScannerImage, types.ImagePullOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to pull scanner image: %w", err)
+	}
+	// ImagePull function's description: It's up to the caller to handle the io.ReadCloser and close it properly.
+	_, _ = io.Copy(io.Discard, pl)
+	_ = pl.Close()
+
+	networkResp, err := c.dockerClient.NetworkCreate(
+		ctx,
+		config.AssetScanID,
+		types.NetworkCreate{
+			CheckDuplicate: true,
+			Driver:         "bridge",
+		},
+	)
+	if err != nil {
+		return "", fmt.Errorf("failed to create scan network: %w", err)
+	}
+
+	scanConfigFilePath := getScanConfigFileName(config)
+	containerResp, err := c.dockerClient.ContainerCreate(
+		ctx,
+		&container.Config{
+			Image: config.ScannerImage,
+			Entrypoint: []string{
+				"/app/vmclarity-cli",
+				"--config",
+				"/tmp/" + filepath.Base(scanConfigFilePath),
+				"--server",
+				config.VMClarityAddress,
+				"--asset-scan-id",
+				config.AssetScanID,
+			},
+		},
+		&container.HostConfig{
+			Binds: []string{fmt.Sprintf("%s:/tmp", path.Dir(scanConfigFilePath))},
+			Mounts: []mount.Mount{
+				{
+					Type:   mount.TypeVolume,
+					Source: config.AssetScanID,
+					Target: MountPointPath,
+				},
+			},
+		},
+		&network.NetworkingConfig{
+			EndpointsConfig: map[string]*network.EndpointSettings{
+				config.AssetScanID: {
+					NetworkID: networkResp.ID,
+				},
+			},
+		},
+		nil,
+		config.AssetScanID,
+	)
+	if err != nil {
+		return "", fmt.Errorf("failed to create scan container: %w", err)
+	}
+
+	return containerResp.ID, nil
+}
+
+func (c *Client) getContainerIdFromContainerName(ctx context.Context, scanName string) (string, error) {
+
+	containers, err := c.dockerClient.ContainerList(ctx, types.ContainerListOptions{
+		All:     true,
+		Filters: filters.NewArgs(filters.Arg("name", scanName)),
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to list containers: %w", err)
+	}
+	if len(containers) == 0 {
+		return "", fmt.Errorf("scan container not found: %w", err)
+	}
+	if len(containers) > 1 {
+		return "", fmt.Errorf("found more than one scan container: %w", err)
+	}
+	return containers[0].ID, nil
+}
+
+func (c *Client) getNetworkIdFromNetworkName(ctx context.Context, scanName string) (string, error) {
+
+	networks, err := c.dockerClient.NetworkList(ctx, types.NetworkListOptions{
+		Filters: filters.NewArgs(filters.Arg("name", scanName)),
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to list networks: %w", err)
+	}
+	if len(networks) == 0 {
+		return "", fmt.Errorf("scan network not found: %w", err)
+	}
+	if len(networks) > 1 {
+		return "", fmt.Errorf("found more than one scan network: %w", err)
+	}
+	return networks[0].ID, nil
+}
+
+func getScanConfigFileName(config *provider.ScanJobConfig) string {
+	// TODO (adamtagscherer): check if os.TempDir() doesn't create a new unique directory every time it's being called
```

Comment:
Should be `getScanConfigFilePath` rather than FileName.

@adamtagscherer `os.TempDir` does not create any dir, just returns the current os-specific temp dir path.
Note that the temp dir path might not be an absolute path (or a valid/existing path at all), so this might need to be handled too.

Reply:
We can keep it like this for now, but it's worth covering this scenario too